web/src/stream_data.ts

```diff
@@ -572,50 +622,54 @@ export function can_post_messages_in_stream(stream) {
         return true;
     }
 
-    if (stream.stream_post_policy === stream_post_policy_values.admins.code) {
+    if (stream.stream_post_policy === (stream_post_policy_values.admins.code as StreamPostPolicy)) {
         return false;
     }
 
     if (page_params.is_moderator) {
         return true;
     }
 
-    if (stream.stream_post_policy === stream_post_policy_values.moderators.code) {
+    if (
+        stream.stream_post_policy ===
+        (stream_post_policy_values.moderators.code as StreamPostPolicy)
+    ) {
         return false;
     }
 
     if (
         page_params.is_guest &&
-        stream.stream_post_policy !== stream_post_policy_values.everyone.code
+        stream.stream_post_policy !== (stream_post_policy_values.everyone.code as StreamPostPolicy)
     ) {
         return false;
     }
 
     const person = people.get_by_user_id(people.my_current_user_id());
-    const current_datetime = new Date(Date.now());
-    const person_date_joined = new Date(person.date_joined);
+    const current_datetime = new Date(Date.now()).getTime();
+    const person_date_joined = new Date(person.date_joined).getTime();
     const days = (current_datetime - person_date_joined) / 1000 / 86400;
     if (
-        stream.stream_post_policy === stream_post_policy_values.non_new_members.code &&
+        stream.stream_post_policy ===
+            (stream_post_policy_values.non_new_members.code as StreamPostPolicy) &&
         days < page_params.realm_waiting_period_threshold
     ) {
         return false;
     }
     return true;
 }
 
-export function is_subscribed_by_name(stream_name) {
+export function is_subscribed_by_name(stream_name: string): boolean | undefined {
     const sub = get_sub(stream_name);
-    return sub !== undefined && sub.subscribed;
+    return sub?.subscribed;
```

Comment:
It’s inconsistent to do this to `is_subscribed_by_name`, `is_subscribed`, `is_web_public` but not `is_invite_only_by_stream_name`, `is_web_public_by_stream_name`.

It seems preferable to return `boolean` here rather than force callers to deal with `boolean | undefined`.